### PR TITLE
Document Fetch, Access and related types for bevy_ecs

### PR DESCRIPTION
**Pull request by alice-i-cecile**
# Objective

- These types are important to bevy_ecs devs and not self-explanatory.

## Solution

- Document them!

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
By @DJMcNab from #3366 on `Archetype`.

> With sparse set components, these can contain many sets of component, although each set is uniquely assigned to a single archetype.

I'm not entirely sure how to explain this cleanly, so help would be appreciated.

File: crates/bevy_ecs/src/query/fetch.rs [hunk 1]
```diff
@@ -232,7 +246,7 @@ impl<T: Component> WorldQuery for &T {
     type ReadOnlyFetch = ReadFetch<T>;
 }
 
-/// The [`FetchState`] of `&T`.
+/// The [`FetchState`] of `&T`, when used as a [Component] in a [WorldQuery].
```
Comment (Weibye) [inline crates/bevy_ecs/src/query/fetch.rs:249]: Is there a difference between `[StructDefinition]` and ```[`StructDefinition`]``` when it comes to what is being linked / how it is rendered in the rust-docs?

**Review by Weibye — APPROVED** [review]
Thanks, this is great and very helpful to someone not already familiar with all of this

File: crates/bevy_ecs/src/query/fetch.rs (hunk 1 above)
Comment (alice-i-cecile) [reply crates/bevy_ecs/src/query/fetch.rs:249]: The latter will be shown in code-style monospace text.

Comment (Weibye) [reply crates/bevy_ecs/src/query/fetch.rs:249]: Gotcha, thanks!

**Comment by alice-i-cecile** [comment]
Closing; I don't care enough to fix the merge conflicts. These can be rewritten in further PRs.

**Pull request closed without merging**